### PR TITLE
libunwind: add versions 1.6.2 and 1.6-stable

### PR DESCRIPTION
**Pull request by mwkrentel**

### CODE REVIEW

File: var/spack/repos/builtin/packages/libunwind/package.py [hunk 1]
```diff
@@ -62,10 +64,10 @@ class Libunwind(AutotoolsPackage):
...
     # The libunwind releases contain the autotools generated files,
     # but the git repo snapshots do not.
-    depends_on('autoconf', type='build', when='@master,1.5-stable')
-    depends_on('automake', type='build', when='@master,1.5-stable')
-    depends_on('libtool',  type='build', when='@master,1.5-stable')
-    depends_on('m4',       type='build', when='@master,1.5-stable')
+    depends_on('autoconf', type='build', when='@1.5-stable,1.6-stable:')
+    depends_on('automake', type='build', when='@1.5-stable,1.6-stable:')
+    depends_on('libtool',  type='build', when='@1.5-stable,1.6-stable:')
+    depends_on('m4',       type='build', when='@1.5-stable,1.6-stable:')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/libunwind/package.py]: Why was `master` removed from the list?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/libunwind/package.py (hunk 1 above)
Comment (alalazo) [reply var/spack/repos/builtin/packages/libunwind/package.py]: @haampie pointed me to the `:` in this `when` argument, that I missed on first read. As currently written this is a range that includes also point releases in the 1.6.X series, right? Is this wanted?

Comment (mwkrentel) [reply var/spack/repos/builtin/packages/libunwind/package.py]: @alalazo It's not removed.  Look closely, it says `1.6-stable:` with the colon, 
and master > 1.6-stable.

Comment (mwkrentel) [reply var/spack/repos/builtin/packages/libunwind/package.py]: @alalazo No, it does not include the point releases.
A couple weeks ago, I added `stable` as an infinity version.
So, 1.6.2 < 1.6-stable < master.

For example, `spack info` reports:
```
Safe versions:  
    master        [git] https://github.com/libunwind/libunwind on branch master
    1.6-stable    [git] https://github.com/libunwind/libunwind on branch v1.6-stable
    1.6.2         http://download.savannah.gnu.org/releases/libunwind/libunwind-1.6.2.tar.gz
    1.5-stable    [git] https://github.com/libunwind/libunwind on branch v1.5-stable
    1.5.0         http://download.savannah.gnu.org/releases/libunwind/libunwind-1.5.0.tar.gz
```

**Comment by mwkrentel** [comment]
@alalazo Can you figure out why the pipeline failed?  
I have a possible theory.  For me, libunwind from savannah.gnu.org was
hard to download via spack.  It fetches just fine with wget or curl,
but not from the python library.

What can I do about that?
Is `download.savannah.gnu.org` part of a larger mirror?
I've had trouble with them before.

**Comment by alalazo** [comment]
> Can you figure out why the pipeline failed?

Let me rebase and see what happens.

File: var/spack/repos/builtin/packages/libunwind/package.py (hunk 1 above)
Comment (mwkrentel) [reply var/spack/repos/builtin/packages/libunwind/package.py]: In part, I added `stable` specifically for libunwind.
Their branches are named stable, so I wanted the spack name to match.

Comment (haampie) [reply var/spack/repos/builtin/packages/libunwind/package.py]: It's still a bad idea, cause it matches 1.7.0. IMHO dependencies should be written in a way that's future proof, and it's easy to fix things now instead when 1.7.0 is released.

Comment (alalazo) [reply var/spack/repos/builtin/packages/libunwind/package.py]: Ugh, so there are 3 gotchas that people need to know to understand this:
1. `master` is an infinity version
2. `stable` is an infinity version
3. Spack versioning system treats `.` as it was `-` when generating the tuple used for comparison

Can we go back to having only fixed releases in this list, so we don't require users to know the above gotchas when trying to understand what will have `m4` & Co. as build dependencies?

**Comment by mwkrentel** [comment]
@alalazo   Ok, I rewrote the when clause.  I was simply trying to write something
that was a little shorter, that's all.  I kinda feel like I'm being
bikeshedded a little.

@haampie  note: when 1.7.0 is added, so will 1.7-stable.  So, it's not
really possible to write a when clause that looks to the future,
at least not until glob patterns and `when='@*-stable,master'`
are allowed.

Anyway, now let's see if we get past the pipeline.
Anything else?

**Review by alalazo — APPROVED** [review]

**Pull request merged**